packages/dm-core-plugins/src/form/fields/ObjectField.tsx

```diff
@@ -85,24 +89,38 @@ const AddObject = (props: { type: string; namePath: string }) => {
       shouldDirty: true,
       shouldTouch: true,
     }
-
+    if (!defaultValue) {
+      dmssAPI
+        .instantiateEntity({
+          entity: { type: type as string },
+        })
+        .then((newEntity: AxiosResponse<any>) =>
+          addDocument(newEntity.data, options)
```

Comment:
If you moved the `options` object on L87 out of handleAdd and into addDocument, you wouldn't have to pass it into AddDocument, and then you wouldn't have to add a type for it either. Also, the object would be declared closer to where it is used

Reply:
Yes, very true!